tensorflow/python/ops/math_ops.py

```diff
@@ -1788,18 +1788,24 @@ def _AddNShape(op):
 def _SelectShape(op):
```

Comment:
You also have to update the C++ shape function in the array_ops.cc definition of the Select op